src/connector/src/parser/protobuf/parser.rs

```diff
@@ -227,7 +240,142 @@ fn detect_loop_and_push(trace: &mut Vec<String>, fd: &FieldDescriptor) -> Result
     Ok(())
 }
 
-pub fn from_protobuf_value(field_desc: &FieldDescriptor, value: &Value) -> Result<Datum> {
+fn extract_any_info(dyn_msg: &DynamicMessage) -> (String, Value) {
+    debug_assert!(
+        dyn_msg.fields().count() == 2,
+        "Expected only two fields for Any Type MessageDescriptor"
+    );
+
+    let type_url = dyn_msg
+        .get_field_by_name("type_url")
+        .expect("Expect type_url in dyn_msg")
+        .to_string()
+        .split('/')
+        .nth(1)
+        .map(|part| part[..part.len() - 1].to_string())
+        .unwrap_or_default();
+
+    let payload = dyn_msg
+        .get_field_by_name("value")
+        .expect("Expect value (payload) in dyn_msg")
+        .as_ref()
+        .clone();
+
+    (type_url, payload)
+}
+
+/// TODO: Resolve the potential naming conflict in the map
+/// i.e., If the two anonymous type shares the same key (e.g., "Int32"),
+/// the latter will overwrite the former one in `serde_json::Map`.
+/// Possible solution, maintaining a global id map, for the same types
+/// In the same level of fields, add the unique id at the tail of the name.
+/// e.g., "Int32.1" & "Int32.2" in the above example
+fn recursive_parse_json(
+    fields: &[Datum],
+    full_name_vec: Option<Vec<String>>,
+    full_name: Option<String>,
+) -> serde_json::Value {
+    // Note that the key is of no order
+    let mut ret: serde_json::Map<String, serde_json::Value> = serde_json::Map::new();
+
+    // The hidden type hint for user's convenience
+    // i.e., `"_type": message.full_name()`
+    if let Some(full_name) = full_name {
+        ret.insert("_type".to_string(), serde_json::Value::String(full_name));
+    }
+
+    for (idx, field) in fields.iter().enumerate() {
+        let mut key;
+        if let Some(k) = full_name_vec.as_ref() {
+            key = k[idx].to_string();
+        } else {
+            key = "".to_string();
+        }
+
+        match field.clone() {
+            Some(ScalarImpl::Int16(v)) => {
+                if key.is_empty() {
+                    key = "Int16".to_string();
+                }
```

Comment:
When encountering any type holding a "nested" struct, we could not get the actual field name inside the struct.
e.g., For the proto below, we could only get the `type_url` and `value` for the `second` and `third` field in `StringStringInt32Value`, rather than the actual field name in those structs. In this case, the current workaround is use anonymous type name as the key, that's when the `full_name_vec` turns `None.

```protobuf
message TestAny {
  int32 id = 1;
  google.protobuf.Any any_value = 2;
}
message StringStringInt32Value {
  string first = 1;
  StringInt32Value second = 2;
  Float32StringValue third = 3;
}
```